src/adapter/graphql/src/queries/datasets/dataset_flow_configs.rs

```diff
@@ -0,0 +1,198 @@
+
+        let flow_config_service = from_catalog::<dyn FlowConfigurationService>(ctx).unwrap();
+        let maybe_flow_config = flow_config_service
+            .find_configuration(
+                FlowKeyDataset::new(self.dataset_handle.id.clone(), dataset_flow_type.into())
+                    .into(),
+            )
+            .await
+            .int_err()?;
+
+        Ok(maybe_flow_config.map(|flow_config| flow_config.into()))
+    }
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+#[derive(Enum, Copy, Clone, Eq, PartialEq)]
+#[graphql(remote = "kamu_flow_system::DatasetFlowType")]
+pub enum DatasetFlowType {
+    Ingest,
+    ExecuteQuery,
+    Compaction,
+}
+
+#[derive(SimpleObject, Debug, Clone, PartialEq, Eq)]
+pub struct FlowConfiguration {
+    pub paused: bool,
+    pub schedule: Option<FlowConfigurationSchedule>,
+    pub batching: Option<FlowConfigurationBatching>,
+}
+
+impl From<kamu_flow_system::FlowConfigurationState> for FlowConfiguration {
+    fn from(value: kamu_flow_system::FlowConfigurationState) -> Self {
+        Self {
+            paused: !value.is_active(),
+            batching: if let FlowConfigurationRule::StartCondition(condition) = &value.rule {
+                Some(FlowConfigurationBatching {
+                    throttling_period: condition.throttling_period.map(|tp| tp.into()),
+                    minimal_data_batch: condition.minimal_data_batch,
+                })
+            } else {
+                None
+            },
+            schedule: if let FlowConfigurationRule::Schedule(schedule) = value.rule {
+                match schedule {
+                    Schedule::TimeDelta(time_delta) => Some(FlowConfigurationSchedule::TimeDelta(
+                        time_delta.every.into(),
+                    )),
+                    Schedule::CronExpression(cron) => {
+                        Some(FlowConfigurationSchedule::Cron(cron.into()))
+                    }
+                }
+            } else {
+                None
+            },
+        }
+    }
+}
+
+#[derive(Union, Debug, Clone, PartialEq, Eq)]
+pub enum FlowConfigurationSchedule {
+    TimeDelta(TimeDelta),
+    Cron(CronExpression),
+}
+
+#[derive(SimpleObject, Debug, Clone, PartialEq, Eq)]
+pub struct FlowConfigurationBatching {
+    pub throttling_period: Option<TimeDelta>,
+    pub minimal_data_batch: Option<i32>,
+}
+
+/////////////////////////////////////////////////////////////////////////////////
+
+#[derive(SimpleObject, Debug, Clone, PartialEq, Eq)]
+pub struct CronExpression {
+    pub cron_expression: String,
+}
+
+impl From<ScheduleCronExpression> for CronExpression {
+    fn from(value: ScheduleCronExpression) -> Self {
+        Self {
+            cron_expression: value.cron_expression,
+        }
+    }
+}
+
+/////////////////////////////////////////////////////////////////////////////////
+
+#[derive(SimpleObject, Debug, Clone, PartialEq, Eq)]
+pub struct TimeDelta {
```

Comment:
Would it make sense to create a separate file for these simple objects / enums in our `scalars` directory?

Reply:
Created for output part, but it's not very clear where to put `InputObject` models.